src/EventRegistry.cc

```diff
@@ -146,4 +147,61 @@ void EventRegistry::ActivateAllHandlers()
 		}
 	}
 
+EventGroupPtr EventRegistry::RegisterGroup(EventGroupKind kind, std::string_view name)
+	{
+	auto key = std::pair{kind, std::string{name}};
+	if ( const auto& it = event_groups.find(key); it != event_groups.end() )
+		return it->second;
+
+	auto group = std::make_shared<EventGroup>(kind, name);
+	return event_groups.emplace(key, group).first->second;
+	}
+EventGroupPtr EventRegistry::LookupGroup(EventGroupKind kind, std::string_view name)
+	{
+	auto key = std::pair{kind, std::string{name}};
+	if ( const auto& it = event_groups.find(key); it != event_groups.end() )
+		return it->second;
+
+	return nullptr;
+	}
+
+EventGroup::EventGroup(EventGroupKind kind, std::string_view name) : kind(kind), name(name) { }
+
+EventGroup::~EventGroup() noexcept { }
+
+namespace
+	{
+// Run through all event handler functions and update their bodies after
+// a group was enabled/disabled.
+//
+// This isn't super efficient as most bodies will not have changed, but it
+// shouldn't take ages either.
+void UpdateFuncBodies()
+	{
+	for ( const auto& handler_name : event_registry->AllHandlers() )
+		{
+		EventHandler* eh = event_registry->Lookup(handler_name);
+		eh->GetFunc()->UpdateBodies();
+		}
+	}
+	}
```

Comment:
Agree it wouldn't take ages :-), but it still feels like doing too much here given that we're just changing a single group. I think I'd add the additional bookkeeping to make the registry track which events handlers are part of each group. Then it can address them directly.

Reply:
I've kept this for now. We could track FuncPtr's in groups, but that's essentially "just" an optimization (likely an effective one though). But it doesn't change the fact that we still need to keep track of groups within each body.